### DISCUSSION TITLE
Preprocess ToC???

### DISCUSSION THREAD

**Discussion by gjblajian** [dolanmiu/docx#2102] [Ideas / Requests]
@dolanmiu, we are having some issues sending our docx generated documents to a third party because the ToC's have not rendered. SyncFusion's library [https://github.com/SyncfusionExamples/Word-TOC-Examples/blob/master/Create-TOC/Console-App-.NET-Core/Create-TOC/Program.cs] can somehow preprocess the ToC such that you are not prompted to update fields when you open the document. After a cursory look at their code in Visual Studio, it is clear they are tracking the dictionary for the ToC in the WordDocument class but their processing of the ToC when calling `document.UpdateTableOfContents();` does not seem to be reliant on .NET. I was wondering if you might be able to replicate this capability in docx?

**Comment by gjblajian** [comment]
updateTableOfContents ultimately calls updateTableOfContent in the attached file
[WordDocument.zip](https://github.com/dolanmiu/docx/files/11344821/WordDocument.zip)

**Reply by dolanmiu** [reply]
Basically how I understand it is that the OOXML standard don't actually have pages. Pages are a word processor concept, like Word of LibreOffice.

This means to get the ToC, you'd need to "render" the ooxml to find the page it sits in, then update the ToC. This is why word processors like Word or Google Docs have an "Update Table of Contents" button, cos it does literally that.

So yeah, it to me sounds like a lot of work. Got to basically count the number of lines in a document, and then find how many lines can fit in a page, then calculate accordingly
